### PR TITLE
dxSpeedDialAction - Fix widget events

### PR DESCRIPTION
**Pull request by groshenkovamarina**

### CODE REVIEW

File: js/ui/speed_dial_action/speed_dial_action.js [hunk 1]
```diff
@@ -99,6 +99,15 @@ const SpeedDialAction = Widget.inherit({
         }
     },
 
+    _createActionByOption(optionName, config) {
+        const emptyFunc = () => {
```
Comment (babich-a) [inline js/ui/speed_dial_action/speed_dial_action.js]: We have something, that called 'noop'. Can we use it here?

File: js/ui/speed_dial_action/speed_dial_item.js [hunk 2]
```diff
@@ -68,6 +82,22 @@ const SpeedDialItem = Overlay.inherit({
...
     },
 
+    _getActionComponent() {
+        return this.option("actionComponent") || this.option("actions")[0];
+    },
+
+    _renderEvents() {
+        this._actionEvents.forEach((actionEvent) => {
+            this.on(actionEvent, () => {
+                this._getActionComponent()._createActionByOption("on" + actionEvent)(
```
Comment (babich-a) [inline js/ui/speed_dial_action/speed_dial_item.js]: We pass 'oninitialized' here but check 'onInitialized' here https://github.com/DevExpress/DevExtreme/pull/8884/files#diff-645315b92c27d92aebe52d8df754b0e0R104 (lower case in the first case and the camel in the second). This is right?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 3]
```diff
@@ -339,4 +338,54 @@ QUnit.module("check action buttons click args", (hooks) => {
...
+        $("#fab-one").dxSpeedDialAction("instance").dispose();
+        $("#fab-two").dxSpeedDialAction("instance").dispose();
+    }),
+
+    test("trigger and args", (assert) => {
+        $("#fab-one")
+            .dxSpeedDialAction()
+            .dxSpeedDialAction("instance")
+            .on("contentReady", (e) => {
+                assert.ok(e.component, "first SDA content ready component in args");
```
Comment (babich-a) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Is it enough to ensure that these properties are defined?

**Comment by BingoRUS** [comment]
@groshenkovamarina let's fix an issue with failed tests.

File: js/ui/speed_dial_action/speed_dial_action.js [hunk 4]
```diff
@@ -99,6 +100,10 @@ const SpeedDialAction = Widget.inherit({
         }
     },
 
+    _createActionByOption(optionName, config, isExecute) {
```
Comment (jsdmitry) [inline js/ui/speed_dial_action/speed_dial_action.js:101]: Maybe make simpler:
```
_createActionByOption(optionName, config, isExecute) {
	if(!!isExecute || (optionName !== "onInitialized" && optionName !== "onDisposing")) {
		return this.callBase(optionName, config);
	}
	return noop;
},
```

File: js/ui/speed_dial_action/speed_dial_item.js [hunk 5]
```diff
@@ -13,6 +13,14 @@ const FAB_ICON_CLASS = "dx-fa-button-icon";
 const OVERLAY_CONTENT_SELECTOR = ".dx-overlay-content";
 
 const SpeedDialItem = Overlay.inherit({
```
Comment (jsdmitry) [inline js/ui/speed_dial_action/speed_dial_item.js]: Use Es6 Class

File: js/ui/speed_dial_action/speed_dial_item.js [hunk 6]
```diff
@@ -81,18 +107,24 @@ const SpeedDialItem = Overlay.inherit({
...
         eventsEngine.off(overlayContent, eventName);
         eventsEngine.on(overlayContent, eventName, (e) => {
-            this._clickAction({ event: e, element: this.$element() });
+            this._clickActionArgs = {
+                event: e,
+                actionElement: this.element(),
+                element: this._getActionComponent().$element()
+            };
+
+            this._clickAction(this._clickActionArgs);
```
Comment (jsdmitry) [inline js/ui/speed_dial_action/speed_dial_item.js]: Why you are defined `_clickActionArgs ` if it's used only once?

File: js/ui/speed_dial_action/speed_dial_action.js (hunk 4 above)
Comment (groshenkovamarina) [reply js/ui/speed_dial_action/speed_dial_action.js:101]: It seems quite simple now

File: js/ui/speed_dial_action/speed_dial_item.js (hunk 5 above)
Comment (groshenkovamarina) [reply js/ui/speed_dial_action/speed_dial_item.js]: Fixed

File: js/ui/speed_dial_action/speed_dial_item.js (hunk 6 above)
Comment (groshenkovamarina) [reply js/ui/speed_dial_action/speed_dial_item.js]: Fixed

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js (hunk 3 above)
Comment (groshenkovamarina) [reply testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Added more checks

**Review by jsdmitry — APPROVED** [review]

**Review by babich-a — APPROVED** [review]

**Pull request merged**